### DISCUSSION TITLE
How to `git push --force`?

### DISCUSSION THREAD

**Discussion by marcinjahn** [jj-vcs/jj#2963] [Q&A] [answered]
I rebased a branch which already had been pushed to remote, now I need to `git push --force` that branch.
When I do `jj git push --force` I get:

> error: unexpected argument '--force' found
> Usage: jj git push [OPTIONS]

How to do that properly?

**Comment by ilyagr** [comment, marked as answer]
Most likely, you need to `jj git fetch` from that remote and resolve any branch conflicts (see the FAQ) if they occur. Then, `jj git push` should work.

**Update:** Perhaps this question should also be in the FAQ.

**Reply by ilyagr** [reply]
I may have also misunderstood. Have you tried a simple `jj git push`? It does an analogue of `git push --force-with-lease` already. What I said before is only relevant if the branch was moved to different places, both locally and on the remote.

**Reply by marcinjahn** [reply]
Yeah, I did try `jj git push` initially, that failed though. I think you are right about the branch conflict, I saw a message about that at some point (don't remember when exactly though), and probably I did not handle that properly. Will do next time! Thanks!
